src/handler.ts

```diff
@@ -223,22 +231,30 @@ export
 class FileHandler extends AbstractFileHandler {
   /**
    * Get file contents given a path.
+   *
+   * #### Notes
+   * Subclasses are free to use any or none of the information in
+   * the model.
    */
-  protected getContents(path: string): Promise<IContentsModel> {
-    return this.manager.get(path, { type: 'file' });
+  protected getContents(model: IContentsModel): Promise<IContentsModel> {
+    return this.manager.get(model.path, { type: model.type });
```

Comment:
Since you specifically have a codemirror editor, you should probably request `type: 'file'` still.  Codemirror won't know what to do with a directory or notebook type.